velox/substrait/tests/PlanConversionTest.cpp

```diff
@@ -0,0 +1,510 @@
+      u_int32_t partitionIndex = planConverter->getPartitionIndex();
+      std::vector<std::string> paths;
+      planConverter->getPaths(paths);
+      // In test, need to get the absolute path of the generated ORC file.
+      std::vector<std::string> absolutePaths;
+      absolutePaths.reserve(paths.size());
+      for (auto path : paths) {
+        std::string currentPath = fs::current_path().c_str();
+        std::string absolutePath = "file://" + currentPath + path;
+        absolutePaths.emplace_back(absolutePath);
+      }
+      std::vector<u_int64_t> starts;
+      planConverter->getStarts(starts);
+      std::vector<u_int64_t> lengths;
+      planConverter->getLengths(lengths);
+      // Construct the result iterator.
+      auto resIter = std::make_shared<WholeComputeResultIterator>(
+          planNode, partitionIndex, absolutePaths, starts, lengths);
+      return resIter;
+    }
+  };
+
+  // This method can be used to create a Fixed-width type of Vector without Null
+  // values.
+  template <typename T>
+  VectorPtr createSpecificScalar(
+      size_t size,
+      std::vector<T> vals,
+      facebook::velox::memory::MemoryPool& pool) {
+    facebook::velox::BufferPtr values = AlignedBuffer::allocate<T>(size, &pool);
+    auto valuesPtr = values->asMutableRange<T>();
+    facebook::velox::BufferPtr nulls = nullptr;
+    for (size_t i = 0; i < size; ++i) {
+      valuesPtr[i] = vals[i];
+    }
+    return std::make_shared<facebook::velox::FlatVector<T>>(
+        &pool, nulls, size, values, std::vector<BufferPtr>{});
+  }
+
+  // This method can be used to create a String type of Vector without Null
+  // values.
+  VectorPtr createSpecificStringVector(
+      size_t size,
+      std::vector<std::string> vals,
+      facebook::velox::memory::MemoryPool& pool) {
+    auto vector = BaseVector::create(VARCHAR(), size, &pool);
+    auto flatVector = vector->asFlatVector<StringView>();
+
+    size_t childSize = 0;
+    std::vector<int64_t> lengths(size);
+    size_t nullCount = 0;
+    for (size_t i = 0; i < size; ++i) {
+      auto notNull = true;
+      vector->setNull(i, !notNull);
+      auto len = vals[i].size();
+      lengths[i] = len;
+      childSize += len;
+    }
+    vector->setNullCount(0);
+
+    BufferPtr buf = AlignedBuffer::allocate<char>(childSize, &pool);
+    char* bufPtr = buf->asMutable<char>();
+    char* dest = bufPtr;
+    for (size_t i = 0; i < size; ++i) {
+      std::string str = vals[i];
+      const char* chr = str.c_str();
+      auto length = str.size();
+      memcpy(dest, chr, length);
+      dest = dest + length;
+    }
+    size_t offset = 0;
+    for (size_t i = 0; i < size; ++i) {
+      if (!vector->isNullAt(i)) {
+        flatVector->set(
+            i, facebook::velox::StringView(bufPtr + offset, lengths[i]));
+        offset += lengths[i];
+      }
+    }
+    return vector;
+  }
+};
+
+// This test will firstly generate mock TPC-H lineitem ORC file. Then, Velox's
```

Comment:
Overall, this test suite has quite a lot of dependencies. Would it be possible to have a leaner test suite with multiple smaller cases exercising only the plan and expr conversion code? 

My point being that we don't necessarily need to execute the full query here, but rather just test that given an input substrait plan/expr, it generates the expect Velox plan/expr. Verifying that Velox correctly executes a given Velox plan/expr is already covered by other test suites. 

We could also build e2e tests like this one, but my suggestion would be to add this along other tpch test execution code.

Reply:
Thanks for your insightful comments.
This is just our first PR. Later, my colleagues and I will submit a series of PRs to improve the UT and add more function support. This UT ensures that the subsequent PRs would not break the functionalities currently supported. We plan to add the UTs you mentioned in our following PRs.